graphix/pauli.py

```diff
@@ -2,441 +2,163 @@
     with complex units and unit constants (with `*`),
     and can be negated.
     """
 
-    def __init__(self, symbol: IXYZ, unit: ComplexUnit):
-        self.__symbol = symbol
-        self.__unit = unit
+    symbol: IXYZ = IXYZ.I
+    unit: ComplexUnit = ComplexUnit.PLUS
+    I: ClassVar[Pauli]
+    X: ClassVar[Pauli]
+    Y: ClassVar[Pauli]
+    Z: ClassVar[Pauli]
 
     @staticmethod
     def from_axis(axis: Axis) -> Pauli:
         """Return the Pauli associated to the given axis."""
-        return Pauli(IXYZ[axis.name], UNIT)
+        return Pauli(IXYZ[axis.name])
 
     @property
     def axis(self) -> Axis:
         """Return the axis associated to the Pauli.
 
         Fails if the Pauli is identity.
         """
-        if self.__symbol == IXYZ.I:
+        if self.symbol == IXYZ.I:
             raise ValueError("I is not an axis.")
-        return Axis[self.__symbol.name]
-
-    @property
-    def symbol(self) -> IXYZ:
-        """Return the symbol (without the complex unit)."""
-        return self.__symbol
+        return Axis[self.symbol.name]
 
     @property
-    def unit(self) -> ComplexUnit:
-        """Return the complex unit."""
-        return self.__unit
-
-    @property
-    def matrix(self) -> npt.NDArray:
+    def matrix(self) -> npt.NDArray[np.complex128]:
         """Return the matrix of the Pauli gate."""
-        return complex(self.__unit) * CLIFFORD[self.__symbol.value + 1]
+        co = complex(self.unit)
+        if self.symbol == IXYZ.I:
+            return co * Ops.I
+        if self.symbol == IXYZ.X:
+            return co * Ops.X
+        if self.symbol == IXYZ.Y:
+            return co * Ops.Y
+        if self.symbol == IXYZ.Z:
+            return co * Ops.Z
+        typing_extensions.assert_never(self.symbol)
 
-    def get_eigenstate(self, eigenvalue=0) -> PlanarState:
+    def eigenstate(self, b: int = 0) -> PlanarState:
         """Return the eigenstate of the Pauli."""
-        from graphix.states import BasicStates
-
+        if b not in {0, 1}:
+            raise ValueError("b must be 0 or 1.")
         if self.symbol == IXYZ.X:
-            return BasicStates.PLUS if eigenvalue == 0 else BasicStates.MINUS
+            return BasicStates.PLUS if b == 0 else BasicStates.MINUS
         if self.symbol == IXYZ.Y:
-            return BasicStates.PLUS_I if eigenvalue == 0 else BasicStates.MINUS_I
+            return BasicStates.PLUS_I if b == 0 else BasicStates.MINUS_I
         if self.symbol == IXYZ.Z:
-            return BasicStates.ZERO if eigenvalue == 0 else BasicStates.ONE
+            return BasicStates.ZERO if b == 0 else BasicStates.ONE
         # Any state is eigenstate of the identity
         if self.symbol == IXYZ.I:
             return BasicStates.PLUS
         typing_extensions.assert_never(self.symbol)
 
-    def __repr__(self) -> str:
-        """Return a fully qualified string representation of the Pauli."""
-        return self.__unit.prefix(f"graphix.pauli.{self.__symbol.name}")
-
```

Comment:
We should keep a `__repr__`: I would expect `- Pauli.I` to be printed as is, `- Pauli.I`, and not `Pauli(symbol=<IXYZ.I: 1>, unit=<ComplexUnit.MINUS: 2>)`.

Reply:
Updated in d19a3639323294e1e16ccec3f9602d5f22f9891a : now `str` returns `-I` and `repr` does `-Pauli.I`, for example.

Reply:
We can make the same remark here as in https://github.com/TeamGraphix/graphix/pull/228#discussion_r1825858161 , but I am not sure there is any strong rule to fix the context where the expression returned by `__repr__` should be valid.